app/registries/foreman/plugin/importer_registry.rb

```diff
@@ -0,0 +1,16 @@
+module Foreman
+  class Plugin
+    class ImporterRegistry
```

Comment:
Nice solution! It is very clean, I like it. One disadvantage though: it's really hard to isolate in tests. Imagine a test that does:

``` ruby

test 'sets default provider' do
  Foreman::Plugin::ImporterRegistry.register :foo, 'FooClass', true

  # some assertions
end
```

Every test that is executed after this one, and relies on default fact importer will fail and it would be really hard to trace back the error.

I prefer @bastilian's [solution](https://github.com/theforeman/foreman/pull/5221/files#diff-e912f1d36ab9771d3896cf47c09a39f6R42). In that solution, I can easily stub registry accessor method with a new instance of the registry and modify it as much as I like. After the test is done, the original accessor will be set and everything will work as usual.